test/otel_telemetry_SUITE.erl

```diff
@@ -28,29 +29,42 @@ init_per_testcase(_, Config) ->
     {ok, _} = application:ensure_all_started(test_app),
     {ok, _} = application:ensure_all_started(opentelemetry_telemetry),
     {ok, _} = application:ensure_all_started(opentelemetry),
-    ot_batch_processor:set_exporter(ot_exporter_pid, self()),
-    otel_telemetry:init(test_app),
+    otel_batch_processor:set_exporter(otel_exporter_pid, self()),
+    otel_telemetry:trace_application(test_app),
+    opentelemetry:register_tracer(test_tracer, "0.1.0"),
     Config.
 
 end_per_testcase(_, Config) ->
     application:stop(telemetry),
     application:stop(telemetry_registry),
-    application:stop(telemetry_app),
+    application:stop(test_app),
     application:stop(opentelemetry_telemetry),
     application:stop(opentelemetry),
     Config.
 
 successful_span(_Config) ->
+    SpanCtx1 = ?start_span(<<"span-1">>),
+    ?set_current_span(SpanCtx1),
     _Result = test_app:handler(ok),
+    ?assertMatch(SpanCtx1, ?current_span_ctx),
```

Comment:
This test does not capture the scenario where during the telemetry span, it starts another span. From what I saw in the implementation, the span started inside the telemetry span won't have the correct parent

Reply:
So it's nested telemetry-based spans?

Reply:
Not exactly, for a specific example think of the following scenario:
- We write a telemetry-based span for Oban using their Telemetry Handlers (let's call it `opentelemetry_oban`)
- A user of `opentelemetry_oban` has a worker that calls `OpenTelemetry.Tracer.with_span/2` (or the equivalent in Erlang)

This test does not capture that kind of scenario, as far as I understood.

Reply:
Test is updated and results in the following trace.
<pre>┌──────────────────────────────────────────────────────────────┐
│            otel_tracer span created by start/end             │
└──────────────────────────────────────────────────────────────┘
  ┌────────────────────────────┐┌─────────────────────────────┐ 
  │  telemetry:span (success)  ││ telemetry:span (exception)  │ 
  └────────────────────────────┘└─────────────────────────────┘ 
    ┌────────────────────────┐                                  
    │telemetry:span (success)│                                  
    └────────────────────────┘                                  </pre>